src/Tolerance/Operation/Runner/RetryPromiseOperationRunner.php

```diff
@@ -95,6 +100,11 @@ protected function onTerminate(PromiseOperation $operation, ThrowableCatcherVote
             try {
                 $this->waitStrategy->wait();
             } catch (WaiterException $waiterException) {
+                // If it is a Guzzle Promise, use Guzzle's objects.
+                if ($promise instanceof PromiseInterface) {
+                    return $fulfilled ? new FulfilledPromise($value) : new RejectedPromise($value);
```

Comment:
This is needed to not break existing code.
Without this change, caller will catch a Tolerance's `PromiseException`. But with the change, the caller will catch a Guzzle's exception (`ClientException`/`ServerException`/`ConnectException`)